src/Services/ElementTabProvider.php

```diff
@@ -0,0 +1,123 @@
+<?php
+namespace DNADesign\Elemental\Services;
+
+use DNADesign\Elemental\Models\BaseElement;
+use Psr\SimpleCache\CacheInterface;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Flushable;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Forms\TabSet;
+
+/**
+ * Provides tab names for
+ */
+class ElementTabProvider implements Flushable
+{
+    use Configurable;
+
+    /**
+     * Set to true to refresh the tab cache on flush
+     *
+     * @config
+     * @var bool
+     */
+    private static $regenerate_on_flush = false;
+
+    /**
+     * @var CacheInterface
+     */
+    protected $cache;
+
+    /**
+     * ElementTabProvider constructor.
+     * @param CacheInterface $cache
+     */
+    public function __construct(CacheInterface $cache = null)
+    {
+        if (!$cache) {
+            $cache = Injector::inst()->get(CacheInterface::class . '.ElementTabCache');
+        }
+        $this->cache = $cache;
+    }
```

Comment:
I do prefer this kind of proper DI, but we should use Injector to handle this instead. You'll need to add the config for it as well as the getter/setter for `$this->cache`

Reply:
Updated - It really doesn't sit well with me that I'm removing this constructor but I understand it's the Injector way...